src/main/java/uk/gov/hmcts/ccd/domain/service/common/CompoundAccessControlService.java

```diff
@@ -344,4 +346,15 @@ private boolean isNullId(JsonNode newItem) {
             || newItem.get("id").equals(NullNode.getInstance())
             || newItem.get("id").asText().equalsIgnoreCase("null");
     }
+
+    private boolean hasAccessControlList(Set<String> userRoles,
+                                         Predicate<AccessControlList> criteria,
+                                         List<AccessControlList> accessControlLists) {
+        // scoop out access control roles based on user roles
+        // intersect and make sure we have access for given criteria
+        return accessControlLists != null && accessControlLists
+            .stream()
+            .filter(acls -> userRoles.contains(acls.getAccessProfile()))
+            .anyMatch(criteria);
+    }
```

Comment:
This looks to be exactly the same implementation as in `AccessControlService` - can we just have it once?

Reply:
removed from here